### PR TITLE
Add Folder exclusion from report

### PR DESCRIPTION
**Pull request by franciscoamado**
:wave: Heyo

I've been using Periphery and noticed that, although we can exclude single files with the `--report-exclude` parameter, it's not possible to exclude folders.

I've implemented it for myself and thought that it might be of interest merging this upstream.

Since I had periphery already running before, I could compare both runs and no increment in runtime was noticed, nor current support for excluding files changed.

It still does not support recursive globs though (as mentioned in docs).

Also, while trying to develop for myself, I've also added rules to the Makefile that allow for testing the developing version of periphery locally (couldn't test the relative path while running in another project, using `swift run periphery`). I believe these could be useful for the development workflow.

Didn't open an issue or discussion prior since the code was already done 😅 

Thanks for putting an effort into this project 👍🏻

TL;DR
* Added a minor change to support excluding folders from the report
* Added 2 Makefile rules to allow testing periphery locally
* Added a minor change to the README explaining this
* Added an entry to the CHANGELOG

### CODE REVIEW

**Comment by ileitch** [comment]
Many thanks for this @franciscoamado. It's actually interesting timing that you made this change, as I was also planning to make some improvements because of #372 & #373. However I think the approach in this PR isn't quite right. Ideally the `Configuration.reportExcludeSourceFiles` function should be updated to support this behavior, and `indexExcludeSourceFiles` too. I will also take a go at implementing recursive globs.

I'm happy to merge in the Makefile changes if you want to split those out into a separate PR. Though FYI, I don't use `swift run periphery` personally, I just point directly at the debug binary, e.g `~/code/py/periphery/.build/debug/periphery`.

**Comment by franciscoamado** [comment]
> Many thanks for this @franciscoamado. It's actually interesting timing that you made this change, as I was also planning to make some improvements because of #372 & #373. However I think the approach in this PR isn't quite right. Ideally the Configuration.reportExcludeSourceFiles function should be updated to support this behavior, and indexExcludeSourceFiles too. I will also take a go at implementing recursive globs.

Completely understand 👍🏻 

> I'm happy to merge in the Makefile changes if you want to split those out into a separate PR. Though FYI, I don't use swift run periphery personally, I just point directly at the debug binary, e.g ~/code/py/periphery/.build/debug/periphery.

didn't realise at the time I could just point to the run the binary 😅  I'll leave it as is 👍🏻

**Comment by ileitch** [comment]
This is now supported in 2.8.0, you can exclude folders with `--report-exclude "Sources/Foo/**"`.
